opengever/maintenance/scripts/fix_broken_sortable_title_indexes.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from plone import api
+from Products.CMFPlone.CatalogTool import MAX_SORTABLE_TITLE
+import logging
+import re
+import transaction
+
+logger = logging.getLogger('opengever.activity')
+SEPARATOR = '-' * 78
+
+# Finds only string containing digits, but was not found by the regex of the
+# plone upgrade (`plone.app.upgrade.v43.alphas.num_sort_regex`)
+num_sort_regex = re.compile('\D+\d+')
+six_digits_regex = re.compile('\d{6}')
+
+
+def fix_broken_sortable_title_indexes(options):
+    """Find brains which has wrongly not been reindexed by
+    the plone.app.upgrade.v43.reindex_sortable_title, and reindex them.
+    """
+
+    # copied from plone.app.upgrade.v43.alphas.py
+    catalog = api.portal.get_tool('portal_catalog')
+    _catalog = catalog._catalog
+    indexes = _catalog.indexes
+    sort_title_index = indexes.get('sortable_title', None)
+    if sort_title_index is None:
+        logger.warn('Fix script cancelled, no sort_title_index found.')
+        return
+
+    from Products.PluginIndexes.FieldIndex import FieldIndex
+    if not isinstance(sort_title_index, FieldIndex.FieldIndex):
+        logger.warn('Fix script cancelled, sort_title_index is not a FieldIndex.')
+        return
+
+    change = []
+    for i, (name, rids) in enumerate(sort_title_index._index.iteritems()):
+        if len(name) > MAX_SORTABLE_TITLE or num_sort_regex.match(name):
```

Comment:
This should still be `.search()` instead of `.match()` - `match` only matches at *the beginning of the string*